src/platform/packages/shared/kbn-tracing/src/init_tracing.ts

```diff
@@ -66,6 +70,14 @@ export function initTracing({
       case 'phoenix':
         LateBindingSpanProcessor.get().register(new PhoenixSpanProcessor(variant.value));
         break;
+
+      case 'grpc':
+        LateBindingSpanProcessor.get().register(new OTLPSpanProcessor(variant.value, 'grpc'));
+        break;
+
+      case 'http':
+        LateBindingSpanProcessor.get().register(new OTLPSpanProcessor(variant.value, 'http'));
+        break;
```

Comment:
TBH, I'm concerned about using a custom implementation of the OTLP Span Processor for the "generic" `grpc` and `http` configurations. The OTLP exporter (gRPC or HTTP) will be used in production to report traces when we migrate away from APM.

Since it will be used in production, I wonder if the processor should be implemented in the generic package `@kbn/tracing`.

Similarly, I'd like to make sure that tweaking it to the inference use case won't be harmful when enabling it in production.

Reply:
Thanks for the response. Glad that you already had this thought out.

TBH, I'm ++ to aligning the config. This is why I wouldn't like to set some expectations that our custom inference span processor might not live up to (if there's no side-effect when enabled in production, I'm happy to keep it as-is).

If the difference between using the custom and the generic span processors is "simply" updating the ES|QL queries, should we consider writing the final queries today (we have 100% of the context, and later might be harder)?